pkg/spi/aws.go

```diff
@@ -29,7 +29,7 @@ func (ms *PluginSPIImpl) NewSession(secret *corev1.Secret, region string) (*sess
 	}
 
 	if workloadIdentityTokenFile, ok := secret.Data["workloadIdentityTokenFile"]; ok {
-		sess, err := session.NewSession()
+		sess, err := session.NewSession(aws.NewConfig().WithRegion(region))
```

Comment:
Is there an advantage in creating a new aws.Config{} object over using the variable `config` defined at the start of this function?

Reply:
I also thought of that but decided to use a separate config since we return a new session at the end anyways https://github.com/gardener/machine-controller-manager-provider-aws/blob/9f7238e49d935544416d2e9a542aca9014751551/pkg/spi/aws.go#L48. I was thinking that I cannot be sure if the `session.NewSession` function would cause any side effect to the passed config so I decided to pass a new one to it.

Reply:
Okay, fair enough
In that case, this is fine